eitprocessing/parameters/tidal_impedance_variation.py

```diff
@@ -0,0 +1,205 @@
+            raise ValueError(msg)
+
+        breaths = self._detect_breaths(continuous_data)
+        return self._calculate_tiv_values(
+            continuous_data.values,
+            continuous_data.time,
+            breaths.values,
+            tiv_method,
+            tiv_timing="continuous",
+        )
+
+    @compute_parameter.register(EITData)
+    def compute_pixel_parameter(
+        self,
+        eit_data: EITData,
+        continuous_data: ContinuousData,
+        sequence: Sequence,
+        tiv_method: Literal["inspiratory", "expiratory", "mean"] = "inspiratory",
+        tiv_timing: Literal["pixel", "continuous"] = "pixel",
+    ) -> np.ndarray:
+        """Compute the tidal impedance variation per breath on pixel level.
+
+        Args:
+            sequence: The sequence containing the data.
+            eit_data: The eit pixel level data to determine the TIV of.
+            continuous_data: The continuous data to determine the continuous data breaths or pixel level inflations.
+            tiv_method: The label of which part of the breath the TIV should be determined on
+                        (inspiratory, expiratory or mean). Defaults to 'inspiratory'.
+            tiv_timing: The label of which timing should be used to compute the TIV, either based on breaths
+                        detected in continuous data ('continuous') or based on pixel inflations ('pixel').
+                        Defaults to 'pixel'.
+
+        Returns:
+            An np.ndarray with the computed TIV values.
+
+        Raises:
+            NotImplementedError: If the method is not 'extremes'.
+            ValueError: If tiv_method is not one of 'inspiratory', 'expiratory', or 'mean'.
+            ValueError: If tiv_timing is not one of 'continuous' or 'pixel'.
+        """
+        if self.method != "extremes":
+            msg = f"Method {self.method} is not implemented."
+            raise NotImplementedError(msg)
+
+        if tiv_method not in ["inspiratory", "expiratory", "mean"]:
+            msg = "tiv_method must be either 'inspiratory', 'expiratory' or 'mean'"
+            raise ValueError(msg)
+
+        if tiv_timing not in ["continuous", "pixel"]:
+            msg = "tiv_timing must be either 'continuous' or 'pixel'"
+            raise ValueError(msg)
+
+        data = eit_data.pixel_impedance
+        _, rows, cols = data.shape
+
+        if tiv_timing == "pixel":
+            pixel_inflations = self._detect_pixel_inflations(
+                eit_data,
+                continuous_data,
+                sequence,
+            )
+            breath_data = pixel_inflations.values
+        else:  # tiv_timing == "continuous"
+            global_breaths = self._detect_breaths(
+                continuous_data,
+            )
+            breath_data = global_breaths.values
+
+        number_of_breaths = breath_data.shape[0] if tiv_timing == "pixel" else len(breath_data)
```

Comment:
Shouldn't the number of breaths always be the same? The number of pixel inflations is the same as the number of breaths, right?

Reply:
Yes they should always give the same result but depending on whether the tiv_timing is pixel or continuous the breath_data is either a list of which the length is the number of breaths or a (x, 32, 32) array of which the first element is the number of breaths. So I implemented this to be able to find the number of breaths regardless of the tiv_timing.

Reply:
Doesn't `len(breath_data)` return `shape[0]` if it's a multidimensional array?

```suggestion
        number_of_breaths = len(breath_data)
```

Reply:
Apparently yes 😅 I have updated it